### ISSUE TITLE
Allow to use within MemoryRouter

### ISSUE THREAD

**Issue by RusinovAnton** [CharlesStover/use-react-router#20]
Hello, thanks for the nice package!

I've just stumbled upon the problem when my component, that using `useReactRouter` hook cannot be tested, I'm getting an error:

```
useReactRouter may only be called within a <Router /> context.
```

For the test environment, we're using `<MemoryRouter />`, and apparently `useReactRouter` doesn't work with its context yet. 

**TL;DR would be nice to add support of MemoryRouter's context for tests**

**Comment by quisido** [comment]
Closing as a `wontfix`.

I'm not able to reproduce this error, as I'm using MemoryRouter in a test environment with BrowserRouter in a production environment.

Given the age of the issue, I'm hoping this was simply fixed by the React Router team themselves when they launched 5.0.x and this package dropped 4.x-beta support.

That said, if it's still a blocking issue for you, bug fixes can be handled by the React Router team's official hook implementation.

Thank you for the engagement! 😊

**Comment by sushruth** [comment]
I had the same issue. This seems to have been resolved when using `react-router@5.1.5`

for Jest environment - note that you may have to write your tests [in this fashion](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/modules/__tests__/useParams-test.js) for better mocking
